python/spatialdata-codec-writer/src/spatialdata_codec_writer/codecs.py

```
@@ -0,0 +1,111 @@
+    plane: np.ndarray, codec: str, encode_options: dict[str, Any]
+) -> bytes | bytearray:
+    array = np.asarray(plane)
+    if codec == CODEC_JPEG2K:
+        return imagecodecs.jpeg2k_encode(array, **encode_options)
+    if codec == CODEC_HTJ2K_OPENJPH:
+        return encode_htj2k_plane_with_options(array, encode_options)
+    raise ValueError(f"Unsupported image codec: {codec}")
+
+
+def chunk_grid(shape: tuple[int, ...], chunks: tuple[int, ...]) -> list[tuple[int, ...]]:
+    ranges = [range((size + chunk - 1) // chunk) for size, chunk in zip(shape, chunks)]
```

Ruff (0.15.17)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)